classes/privacy/provider.php

```diff
@@ -335,8 +363,13 @@ public static function delete_data_for_users(\core_privacy\local\request\approve
                     AND zmp.userid $insql";
 
         $params = array_merge($inparams, ['contextid' => $context->id, 'modlevel' => CONTEXT_MODULE]);
-
-        $DB->delete_records_select('zoom_meeting_participants', "id IN ($sql)", $params);
+        if ($zooms = $DB->get_records_sql($sql, $params)) {
+            $zmpids = [];
+            foreach ($zooms as $zoom) {
+                $zmpids[] = $zoom->id;
+            }
+            $DB->delete_records_select('zoom_meeting_participants', "id IN (".implode(',', $zmpids).")");
+        }
```

Comment:
They we can hopefully avoid pulling all of the data locally using this command:
```suggestion
        $DB->delete_records_select('zoom_meeting_participants', "userid $insql AND detailsid IN ($sql)", $params);
```